### PR TITLE
Add new module - Internal Developer Portal (Beta)

### PR DESCRIPTION
**Pull request by OrkoHunter**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

Adds a new Harness module

Preview URL: https://646f83efa5cf291396f106f9--harness-developer.netlify.app/

### CODE REVIEW

**Comment by ravilach** [comment]
We eventually will also need the Coveo Facet updated cc @wei-harness

**Comment by ravilach** [comment]
@ orkohunter good for us to check this out?

**Comment by OrkoHunter** [comment]
Hi @ravilach - added a one page overview of IDP with a note that this is available in closed beta to existing customers. We have a [long list of docs](https://harness.atlassian.net/wiki/spaces/IDP/pages/21275968000/Customer-facing+documentation+requirements) that are being written by the team. We will continuously add them here. Do you think we can merge this PR to lay the foundations of IDP so that it's easier for others in the team to add their respective docs/tutorials?

Screenshot 2023-05-25 at 8 55 34 PM

**Comment by ravilach** [comment]
@OrkoHunter since we a good set of changes in the docusarus.js and the sidebar.js, mind bringing those up to speed before we merge? CC @KimberlyFields this is ready for review.

File: docs/internal-developer-portal/intro.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+helpdocs_category_id:
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+:::note
+This module is currently available in Beta to limited customers. Contact Harness Support if you would like to try it out.
+:::
+
+Harness IDP lets developers **create** new software components in seconds following the company’s best practices. It lets them **manage** their software by presenting a developer-centric view of all the relevant information. And it lets them **explore** the internal software ecosystem of the company, discover tech docs, APIs, services to enable better collaboration.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/intro.md:15]: I would like to ask a question or two here. I definitely have suggestions for this text.

File: docs/internal-developer-portal/intro.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Fully integrated into the Harness platform, IDP reduces the maintenance overhead and investment required from customers and is enterprise ready with our platform features.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/intro.md:17]: Same as previous: I would like to ask a couple of questions here.

File: src/components/Docs/data/internalDeveloperPortal.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        time: "9min",
+        link: "/tutorials/build-code/build/tfc-notification",
+      },
+    ];
+    */
+
+// Docs
+export const docsCards: CardSections = [
+  {
+    name: "Learn more",
```
Comment (deepakbolangady) [inline src/components/Docs/data/internalDeveloperPortal.ts:54]:
```suggestion
    name: "Get started with IDP",
```

File: src/components/Docs/data/internalDeveloperPortal.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    description: "",
+    list: [
+      {
+        title: "Overview",
+        module: MODULES.platform,
+        description:
+          "Read this to know more about the use-cases and features of Harness IDP.",
```
Comment (deepakbolangady) [inline src/components/Docs/data/internalDeveloperPortal.ts:61]:
```suggestion
          "Learn how Harness's developer-centric portal helps platform engineering and development teams improve productivity."
```

File: docs/internal-developer-portal/intro.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+---
+title: Harness IDP Overview
+description: A home for developers to create, manage and explore software
+sidebar_position: 50
+helpdocs_topic_id:
+helpdocs_category_id:
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/intro.md:10]: # Overview of Harness IDP

**Review by deepakbolangady — CHANGES_REQUESTED** [review]
I need some inputs for the first two paragraphs (we can look at them before beta). I've entered my feedback on the other sections of the document. I'll continue to review this text as more commits are added. Thank you.

File: docs/internal-developer-portal/intro.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+# Overview of Harness IDP
+
+:::note
+This module is currently available in Beta to limited customers. Contact Harness Support if you would like to try it out.
+:::
+
+Harness IDP lets you **create** new software components in seconds adhering to your company’s best practices. It lets you **manage** the software you own by presenting a developer-centric view of all the relevant information like service health, deployments and alerts. And it lets you **explore** the internal software ecosystem of the company, discover technical docs, APIs, services to enable better collaboration.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/intro.md:17]:
```suggestion
Harness IDP enables you to create new software components quickly while adhering to your company’s best practices. It enables you to manage the software you own by presenting a developer-centric view of all relevant information such as service health, deployments, and alerts. It also enables you to explore the internal software ecosystem of the company, discover technical documentation, APIs, and services, all of which enable better collaboration.
```
* Can we back up the "in seconds" claim, or would the time taken depend on variables? I would prefer "Harness IDP enables rapid development of ...", but in the current sentence, "quickly" works too (still subjective but better than "in seconds").
* When someone says "lets you", our thoughts go to permission. "Enables you" is better. It means "gives you the ability to" or "empowers you" in a way. I hope these suggestions can be implemented. 
* Let's run with this description. In the next iteration, let's bring in the personas involved. The first part of this description is for the Platform Engineering team and the last part is for development teams, I think. Both user segments should be able to connect with this text. Thanks.

File: docs/internal-developer-portal/intro.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+Fully integrated into the Harness platform, IDP reduces the maintenance overhead and investment required from the platform engineering and is enterprise ready.
```
Comment (deepakbolangady) [inline docs/internal-developer-portal/intro.md:19]:
```suggestion
Fully integrated into the Harness platform, IDP reduces the maintenance overhead and investment required from platform engineering teams and is enterprise ready.
```

**Review by deepakbolangady — APPROVED** [review]
Approving with a few comments. Thanks.

**Review by deepakbolangady — APPROVED** [review]

**Pull request merged**